src/main/AndroidManifest.xml

```diff
@@ -47,5 +47,13 @@
         android:screenOrientation="portrait"/>
     <activity android:name="SettingsDialogActivity"
         android:screenOrientation="portrait"/>
+    <activity android:name="AppUrlIntentActivity" android:launchMode="singleInstance">
+      <intent-filter>
+        <action android:name="android.intent.action.VIEW" />
+        <category android:name="android.intent.category.DEFAULT" />
+        <category android:name="android.intent.category.BROWSABLE" />
+        <data android:scheme="@string/scheme" android:host="@string/fixed_app_url" android:pathPattern=".*"/>
```

Comment:
This will match any url that points to the app host, instead of just the one with the token login. Thoughts?

Reply:
Definitely a good idea. Does it actually open the right page or just the standard landing page?

Reply:
Opens the right page.

Reply:
I just thought about a potential problem. 
If the URL that the user clicks points to something they can't see or doesn't exist, their app is stuck on that page. The only way to navigate back to webapp would be to click on a "correct" link and overwrite the current navigation. 

I would be nice if we could detect we're not loading HTML and redirect the user straight away, but I have no idea if that's feasible or possible. 

What do you think ?

Reply:
There are two kinds of doesn't exist or not allowed to see... the server kind and the angular kind. The angular kind should just work, so I assume you're talking about the server kind? I don't know about detecting a JSON response in the client, but we could modify the server to respond with an error page if it thinks the request came from a human. The error page could have a link to "home" or "login" or whatever. I'm not too worried about it because it'll be rare. Furthermore it's not specific to this feature - any link to a URL (token login or otherwise) could have this same problem. Maybe raise a new bug for human readable error pages?

Reply:
Yes, I was talking about the server kind, yes. I'll raise an issue.

Reply:
Issue raised https://github.com/medic/cht-core/issues/6546